containers/src/Kokkos_Vector.hpp

```diff
@@ -187,14 +191,18 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
   typename std::enable_if<impl_is_input_iterator<InputIterator>::value,
                           iterator>::type
   insert(iterator it, InputIterator b, InputIterator e) {
+    ptrdiff_t count = std::distance(b, e);
+    if (count == 0) return it;
+
+    if ((_size == 0) && (it == begin())) {
+      resize(count);
```

Comment:
But you resize to `size() + count` so the empty vector is turned into a vector of size `2*counts`.

Reply:
IN that particular point I resize the empty vector in one of length count don't I?

Reply:
Yes, which makes the vector have a size of `count`, and then 10 lines later your resize again which yield `2 * counts`.

Reply:
Still broken you need to return early or change the logic.
```C++
Kokkos::Vector<...> v1(...);  // an empty vector
assert(v1.size() == 0);

Kokkos::Vector<...> v2(...);  // some other "compatible" vector not empty
assert(v2.size() != 0);

v1.insert(v1.end(), v2.begin(), v2.end());
assert(v1.size() == ???); // with your current implementation the answer is "2 * v2.size()" :(
```